[harvester doc] add doc of alertmanager (#209)
harvester/docs: docs/monitoring/monitoring.md

'service': 'longhorn-backend', 'severity': 'warning'},
'commonAnnotations': {'description': 'Longhorn volume is Degraded for more than 5 minutes.', 'runbook_url': 'https://longhorn.io/docs/1.3.0/monitoring/metrics/',
'summary': 'Longhorn volume is Degraded'},
'externalURL': 'https://192.168.122.200/api/v1/namespaces/cattle-monitoring-system/services/http:rancher-monitoring-alertmanager:9093/proxy',
'version': '4',
'groupKey': '{}/{namespace="longhorn-system"}:{}',
'truncatedAlerts': 0
}
```

:::note

Different receivers may present the alerts in different formats. For details, please refer to the related documents.

:::

#### Known Limitation

The `AlertmanagerConfig` is enforced by the `namespace`. Gloabl-level `AlertmanagerConfig` without a namespace is not supported.

We have already created a [GithHb issue](https://github.com/harvester/harvester/issues/2760) to track upstream changes. Once the feature is available, `Harvester` will adopt it.

### View and Manage Alerts

#### From Alertmanager Dashboard

> You can visit the original dashboard of `Alertmanager` from below link.
> 
> > https://the-cluster-vip/api/v1/namespaces/cattle-monitoring-system/services/http:rancher-monitoring-alertmanager:9093/proxy/#/alerts

Reviewer: ```suggestion
You can visit the original dashboard of `Alertmanager` from the link below. Note that you need to replace <cluster-vip> with the actual cluster-vip:

> https://<cluster-vip>/api/v1/namespaces/cattle-monitoring-system/services/http:rancher-monitoring-alertmanager:9093/proxy/#/alerts
```

Reviewer: Thanks for explaining the links! If so, I think it's better to tell users the need to replace the variable.

Author: seems `<cluster-vip>` is not fit here.

![image](https://user-images.githubusercontent.com/31133476/197510628-83224651-0203-47e4-be3a-cf71ef1075bb.png)

Reviewer: Then maybe we should remove the angle brackets ^^

Author: keep `the-cluster-vip` and without `<>` makes CI happy.

The overall view of the `Alertmanager` dashboard is as follows.

![](/img/v1.1/monitoring/alertmanager-dashboard.png)

You can view the details of an alert:

![](/img/v1.1/monitoring/alert-view-detail.png)

#### From Prometheus Dashboard

> You can visit the original dashboard of `Prometheus` from below link.
> 
> > https://the-cluster-vip/api/v1/namespaces/cattle-monitoring-system/services/http:rancher-monitoring-prometheus:9090/proxy/

Reviewer: ```suggestion
You can visit the original dashboard of `Prometheus` from the link below. Note that you need to replace <cluster-vip> with the actual cluster-vip.

> https://<cluster-vip>/api/v1/namespaces/cattle-monitoring-system/services/http:rancher-monitoring-prometheus:9090/proxy/
```

For versions `<= v1.0.0`, the related path and default value are not specified in the `managedchart rancher-monitoring`, you need to add them accordingly.

## Alertmanager

_Available as of v1.1.0_

> `Harvester` has an `Alertmanager` to collect and manage all the alerts happened in the cluster.

Reviewer 2: ```suggestion
`Harvester` has an `Alertmanager` to collect and manage all the alerts happening in the cluster.
```

Author: change to `happened/happening`.

### Alertmanager config

#### Enable/Disable Alertmanager

`Alertmanager` is enabled by default, users may disable it from following config path.

![](/img/v1.1/monitoring/alertmanager-config-enable-and-resource.png)

#### Change resource setting

The resource setting of `Alertmanager` can also be changed. It is shown in previous picture.

#### Config AlertmanagerConfig from WebUI

To send the alerts to third party servers, users need to config the `AlertmanagerConfig`.

> The WebUI config path is `Monitoring & Logging` -> `Monitoring` -> `Alertmanager Configs`.

Reviewer 2: I would rephrase this as 
```suggestion
On the WebUI, navigate to `Monitoring & Logging` -> `Monitoring` -> `Alertmanager Configs`.
```

Create `Alertmanager Config`, click `Namespace` to select the target namespace from drop down menu.

![](/img/v1.1/monitoring/alertmanager-config-create-1.png)

> View the created `Alertmanager Configs`, click one of them to continue config.

Reviewer: Using the one we just created as an example will be clearer. Otherwise, it may be confusing.

![](/img/v1.1/monitoring/view-alertmanager-config.png)

Click `Add Receiver`.

![](/img/v1.1/monitoring/prepare-to-add-receiver.png)

Select the receiver type, for example, `Webhook`. Input reciver name, and click `Add Webhook`.

![](/img/v1.1/monitoring/webhook-receiver-1.png)

Fill the required param and click `Create`.

![](/img/v1.1/monitoring/webhook-receiver-2.png)

#### Config AlertmanagerConfig from CLI

User may also add `AlertmanagerConfig` from CLI.

Exampe: a Webhook type receiver in default namespace.

```
cat << EOF > a-single-receiver.yaml
apiVersion: monitoring.coreos.com/v1alpha1
kind: AlertmanagerConfig
metadata:
  name: amc-example
  # namespace: your value
  labels:
    alertmanagerConfig: example
spec:
  route:
    continue: true
    groupBy:
    - cluster
    - alertname

>     receiver: "amc-webhook-reciver"
>   receivers:
>   - name: "amc-webhook-reciver"

Reviewer: ```suggestion
    receiver: "amc-webhook-receiver"
  receivers:
  - name: "amc-webhook-receiver"
```

Reviewer: Please double-check if these are typo

Author: updated, thanks.

    webhookConfigs:
    - sendResolved: true
      url: "http://192.168.122.159:8090/"
EOF

# kubectl apply -f a-single-receiver.yaml
alertmanagerconfig.monitoring.coreos.com/amc-example created

# kubectl get alertmanagerconfig -A
NAMESPACE   NAME          AGE
default     amc-example   27s

```

#### Example of webhook received alert

> The alert sent to an webhook server will be in following format:

Reviewer 2: Modifying Vicky's suggestion a bit.
```suggestion
Alerts sent to the webhook server will be in the following format:
```

```
{
'receiver': 'longhorn-system-amc-example-amc-webhook-reciver', 
'status': 'firing', 
'alerts': [], 
'groupLabels': {}, 
'commonLabels': {'alertname': 'LonghornVolumeStatusWarning', 'container': 'longhorn-manager', 'endpoint': 'manager', 'instance': '10.52.0.83:9500', 'issue': 'Longhorn volume is Degraded.',
'job': 'longhorn-backend', 'namespace': 'longhorn-system', 'node': 'harv2', 'pod': 'longhorn-manager-r5bgm', 'prometheus': 'cattle-monitoring-system/rancher-monitoring-prometheus',
'service': 'longhorn-backend', 'severity': 'warning'},
'commonAnnotations': {'description': 'Longhorn volume is Degraded for more than 5 minutes.', 'runbook_url': 'https://longhorn.io/docs/1.3.0/monitoring/metrics/',
'summary': 'Longhorn volume is Degraded'},
'externalURL': 'https://192.168.122.200/api/v1/namespaces/cattle-monitoring-system/services/http:rancher-monitoring-alertmanager:9093/proxy',
'version': '4',
'groupKey': '{}/{namespace="longhorn-system"}:{}',
'truncatedAlerts': 0
}
```

:::note

> Different receiver may represent the alert in it's format, please refer the related documents.

Reviewer 2: I would link to the related documents if any within our site.

:::

#### Known limitation

The `AlertmanagerConfig` is enforced by namespace. Gloabl level `AlertmanagerConfig` without namespace is not supported.

One github issues is created to track upstream change, once the feature is available, `Harvester` will adapt it.

https://github.com/harvester/harvester/issues/2760

### View and manage alerts

#### From Alertmanager dashboard

> From below link, user may visit the `Alertmanager` original dashboard.
> 
> > https://the-cluster-vip/api/v1/namespaces/cattle-monitoring-system/services/http:rancher-monitoring-alertmanager:9093/proxy/#/alerts

Reviewer: ```suggestion
You can visit the original dashboard of `Alertmanager` from [here](https://the-cluster-vip/api/v1/namespaces/cattle-monitoring-system/services/http:rancher-monitoring-alertmanager:9093/proxy/#/alerts).
```

Author: Because the link itself is invalid, user need to copy the link and change the `the-cluser-vip` to real vip address (e.g. 192.168.122.100). Here we need to represent the raw link text for user to copy and modify.

> Overall view of `Alertmanager` dashboard.

Reviewer: ```suggestion
The overall view of the `Alertmanager` dashboard is as follows:
```

![](/img/v1.1/monitoring/alertmanager-dashboard.png)

> View the details of an alert.

Reviewer: ```suggestion
You can view the details of an alert:
```

![](/img/v1.1/monitoring/alert-view-detail.png)

#### From Prometheus dashboard

> From below link, user may visit the `Prometheus` original dashboard, which has detailed information of alert.
> 
> > https://the-cluster-vip/api/v1/namespaces/cattle-monitoring-system/services/http:rancher-monitoring-prometheus:9090/proxy/

Reviewer: ```suggestion
You can visit the original dashboard of `Prometheus` from [here](https://the-cluster-vip/api/v1/namespaces/cattle-monitoring-system/services/http:rancher-monitoring-prometheus:9090/proxy/), which offers detailed information about the alert.
```

Author: Similar. Because the link itself is invalid, user need to copy the link and change the `the-cluser-vip` to real vip address (e.g. 192.168.122.100). Here we need to represent the raw link text for user to copy and modify.

> The `Alert` top menu, shows all defined `PrometheusRule`s. With the filter `Inactive`, `Pending`, `Firing`, user can get information quickly.

Reviewer: ```suggestion
The `Alerts` menu in the top navigation bar shows all defined `PrometheusRule`s. You can use the filters `Inactive`, `Pending`, and `Firing` to quickly find the information that you need.
```

Reviewer 2: The `` sign needs to either incorporate the plural form OR you need to rephrase it as below for correctness.

```suggestion
The `Alerts` menu in the top navigation bar shows all defined rules in Prometheus. You can use the filters `Inactive`, `Pending`, and `Firing` to quickly find the information that you need.
```
